utils/sf-config.js

```diff
@@ -7,57 +7,101 @@
 
 const { dirname } = require('path');
 const { cosmiconfigSync } = require('cosmiconfig');
-const { isMultiPackageProject } = require('./project-type');
+const { isPlugin } = require('./project-type');
 
 const PACKAGE_DEFAULTS = {
   scripts: {
-    build: 'sf-build',
+    build: 'wireit',
     clean: 'sf-clean',
     'clean-all': 'sf-clean all',
-    compile: 'sf-compile',
+    compile: 'wireit',
     docs: 'sf-docs',
-    format: 'sf-format',
-    // Cleaner errors than ts-node through tests
-    pretest: 'sf-compile-test',
-    test: 'sf-test',
-    lint: 'sf-lint',
+    format: 'wireit',
+    // this will be removed
+    pretest: undefined,
+    posttest: undefined,
+    test: 'wireit',
+    'test:compile': 'wireit',
+    'test:only': 'wireit',
+    lint: 'wireit',
     prepack: 'sf-prepack',
   },
-};
-
-const LERNA_DEFAULTS = {
-  scripts: {
-    build: 'lerna build',
-    clean: 'lerna clean',
-    'clean-all': 'lerna sf-clean all',
-    compile: 'lerna compile',
-    docs: 'lerna docs',
-    format: 'lerna format',
-    test: 'lerna test',
-    lint: 'lerna lint',
+  wireit: {
+    build: {
+      dependencies: ['compile', 'lint'],
+    },
+    compile: {
+      command: 'tsc -p . --pretty --incremental',
+      files: ['src/**/*.ts', 'tsconfig.json', 'messages/**'],
+      output: ['lib/**', '*.tsbuildinfo'],
+      clean: 'if-file-deleted',
+    },
+    format: {
+      command: 'prettier --write "+(src|test|schemas)/**/*.+(ts|js|json)|command-snapshot.json"',
+    },
+    lint: {
+      command: 'eslint src test --color --cache --cache-location .eslintcache',
+      files: ['src/**/*.ts', 'test/**/*.ts', 'messages/**', '.eslintignore', '.eslintrc.js'],
+      output: [],
+    },
+    'test:compile': {
+      command: 'tsc -p "./test" --pretty',
```

Comment:
Aside from getting prettier errors from `tsc`, is there any reason for this script to exist?

We could just add this env in the `test:only` script and have `ts-node` pretty-print errors:
https://github.com/TypeStrong/ts-node#pretty

`ts-node` still does type-checking (NUTs use `swc` in `bin/dev` and skip type checking)

Reply:
it compiles the NUTs, which `test:only` won't do.

Without that, you could push broken NUTs get a failure way later, from GHA, after the linux UT have run.

It's a small value that probably won't surface often, but also a fairly small cost (no time, maybe slows down the processor just a hair).

Reply:
definitely about to add a comment about it